lib/src/over_react_test/react_util.dart

```diff
@@ -490,25 +489,27 @@ Element queryByTestId(dynamic root, String value, {String key = defaultTestIdKey
 ///     </div>
 ///
 ///     queryAllByTestId(renderedInstance, 'value'); // returns both `inner` `<div>`s
-List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey, bool searchInShadowDom = false, int shadowDepth}) {
+List<Element> queryAllByTestId(dynamic root, String value, {String key = defaultTestIdKey, bool searchInShadowDom = false, int? shadowDepth}) {
   return _findDeep(findDomNode(root), _makeTestIdSelector(value, key: key), searchInShadowDom: searchInShadowDom, findMany: true, depth: shadowDepth);
 }
 
 String _makeTestIdSelector(String value, {String key = defaultTestIdKey}) => '[$key~="$value"]';
 
-List<Element> _findDeep(Node root, String itemSelector, {bool searchInShadowDom = false, bool findMany = true, int depth}) {
+List<Element> _findDeep(Node? root, String itemSelector, {bool searchInShadowDom = false, bool findMany = true, int? depth}) {
   List<Element> nodes = [];
-  void recursiveSeek(Node _root, int _currentDepth) {
+  void recursiveSeek(Node? _root, int _currentDepth) {
     // The LHS type prevents `rootQuerySelectorAll` from returning `_FrozenElementList<JSObject<undefined>>` instead of `<Element>` in DDC
-    final List<Element> Function(String) rootQuerySelectorAll = _root is ShadowRoot ? _root.querySelectorAll : _root is Element ? _root.querySelectorAll : null;
+    final List<Element> Function(String) rootQuerySelectorAll = _root is ShadowRoot ? _root.querySelectorAll : _root is Element ? _root.querySelectorAll : (String s) => [];
```

Comment:
It looks like this would throw before with a null-error if it ever got to that final case; throwing here would make sense to me, since it's an unexpected case that's either impossible, or that this implementation should have handled. Could we preserve that?

Maybe
```dart
    final List<Element> Function(String) rootQuerySelectorAll;
    if ( _root is ShadowRoot) {
      rootQuerySelectorAll = _root.querySelectorAll;
    } else if (_root is Element) {
      rootQuerySelectorAll =  _root.querySelectorAll;
    } else {
      throw Exception('Unhandled node that is neither a ShadowRoot nor an Element: $_root');
    }
```
or maybe just
```dart
    final List<Element> Function(String) rootQuerySelectorAll = _root is ShadowRoot ? _root.querySelectorAll : (_root as Element).querySelectorAll;
```